clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -249,6 +249,19 @@ static bool CheckSYCLKernelName(Sema &S, SourceLocation Loc,
 
   return false;
 }
+void SemaSYCL::CheckSYCLExternalFunctionDecl(FunctionDecl *FD) {
+  for (auto *SEAttr : FD->specific_attrs<SYCLExternalAttr>()) {
```

Comment:
I just happened to notice this while scrolling for Alexey's comments. It isn't necessary to use a `for` loop here since there is no meaningful distinction between one `sycl_external` attribute and another; unlike `sycl_kernel_entry_point`, there is no attribute argument that can cause them to be different. We should be able to just get the attribute.
```suggestion
  const auto *SEAttr = FD->getAttr<SYCLExternalAttr>();
  assert(SEPAttr && "Missing sycl_external attribute");
```